build/v1/types.go

```diff
@@ -1022,23 +1026,51 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// ImageStreamTagReference captures the required elements for identifying the ImageStreamTag referenced by the more
+// generic ObjectReference BuildTriggerPolicy.ImageChange.From.  It is used by ImageChangeTriggerStatus, where a
+// specific instance of ImageChangeTriggerStatus in maintained in BuildConfigStatus.ImageChangeTriggers for each entry
+// in the BuildConfigSpec.Triggers array where the BuildTriggerPolicy.ImageChange pointer is set to a non-nil value
+type ImageStreamTagReference struct {
+	// namespace is the namespace where the ImageStreamTag used for an ImageChangeTrigger is located
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
+
+	// name is the name of the ImageStreamTag used for an ImageChangeTrigger
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+}
+
 // ImageChangeTriggerStatus tracks the latest resolved status of the associated ImageChangeTrigger policy
 // specified in the BuildConfigSpec.Triggers struct.
 type ImageChangeTriggerStatus struct {
-	// lastTriggeredImageID is the sha/id of the imageref cited in the 'from' field the last time this BuildConfig was triggered.
-	// It is not necessarily the sha/id of the image change that triggered a build.
-	// This field is updated for all image change triggers when any of them triggers a build.
+	// lastTriggeredImageID represents, at the last time a Build for this BuildConfig was instantiated, the sha/id of
+	// the image referenced by the the ImageStreamTag cited in the 'from' of this struct.
+	// The lastTriggeredImageID field will be updated by the OpenShift APIServer on all instantiations of a Build from
+	// the BuildConfig it processes, regardless of what is considered the cause of instantiation.
+	// Specifically, an instantiation of a Build could have been manually requested, or could have resulted from
+	// changes with any of the Triggers defined in BuildConfigSpec.Triggers.
+	// The reason for always updating this field across all ImageChangeTriggerStatus instances is to prevent
+	// multiple builds being instantiated concurrently when multiple ImageChangeTriggers fire concurrently.  The system
+	// compares the the sha/id stored here with the associated ImageStreamTag's sha/id for the image.  If they match,
+	// then this trigger is not a valid reason for instantiating a Build.  So when ImageChangeTriggers fire concurrently,
+	// only one of them can "win", meaning selected as the cause for a Build instantiation request.
+	// Lastly, to clarify exactly what is meant by "Build instantiation", from a REST perspective, it is a HTTP POST of a
+	// BuildRequest object as the HTTP Body that is made to the OpenShift APIServer, where that HTTP POST also specifies
+	// the "buildconfigs" resource,  "instantiate" subresource, as well as the namespace and name of the BuildConfig.
```

Comment:
Likewise, too many details for API documentation.

```suggestion
	// lastTriggeredImageID represents the sha/id of the ImageStreamTag used when a Build for this BuildConfig was started.
	// The lastTriggeredImageID will be updated each time a Build for this BuildConfig is started, even if this ImageStreamTag is not the reason the Build is started.
```